src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/DungeonSecrets.java

```diff
@@ -76,6 +79,7 @@ public class DungeonSecrets {
     protected static final Logger LOGGER = LoggerFactory.getLogger(DungeonSecrets.class);
     private static final String DUNGEONS_PATH = "dungeons";
     private static final Path CUSTOM_WAYPOINTS_DIR = SkyblockerMod.CONFIG_DIR.resolve("custom_secret_waypoints.json");
+    private static final Pattern KEY_FOUND = Pattern.compile("^(?<name>\\w+) has obtained (?<type>Blood|Wither) Key!$");
```

Comment:
This regex isn't correct as it doesn't account for ranked players:
[MVP+] AzureAaron has obtained Wither Key!